auth/auth.go

```diff
@@ -36,14 +36,20 @@ func NewClient(clientID string, clientSecret string, issuerURL string, httpClien
 	// Need to wait for identity service to become available.
 	var provider *oidc.Provider
 	var err error
+	var backoff time.Duration = 1 * time.Second
 	for {
 		provider, err = oidc.NewProvider(ctx, issuerURL)
 		if err != nil {
-			log.Errorf("failed to query provider %q: %v", issuerURL, err)
-			time.Sleep(5 * time.Second)
+			time.Sleep(backoff)
 		} else {
+			backoff = 1
 			break
 		}
+		if backoff >= time.Duration(30 * time.Second) {
+			log.Errorf("issue querying the provider: %q. Will retry again in 30 seconds", issuerURL)
```

Comment:
Would it be possible for us to add more context to the error message so that it's easier for customers and support to identify what's going on? Provider is such an overloaded term in our software. Maybe something like

log.Errorf("authentication client failed to contact issuer  %q: %v", issuerURL, err)

Reply:
Maybe something like 

`log.Errorf("authentication client failed to contact issuer %q: %v. will rertry in ~30 seconds", issuerURL, err)`

Reply:
👍

Reply:
Piggybacking on this comment, we need to invert the if statement.

Current behaviour is - 
"try, sleep, message that we're going to sleep, try, sleep" instead of
"try, message that we're going to sleep, sleep, try, ..."